### PR TITLE
don't add generalizedTel definitions to mutual blocks

### LINKED ISSUE #7530: Generalized variable blocks projection-likeness
Consider the following Agda file:
```agda
variable a : Set

data C (a : Set) : Set where

member : C a → a → a
member _ w = w
```
Running agda with `-v tc.proj.like:50` reports the following:
```
checkDecl: checking projection-likeness of [RollbackWindow.C]
RollbackWindow.C is abstract or not a function, thus, not considered for projection-likeness
checkDecl: checking projection-likeness of [RollbackWindow.member, RollbackWindow..generalizedField-a,
 RollbackWindow.GeneralizeTel, RollbackWindow.mkGeneralizeTel]
mutual definitions are not considered for projection-likeness
```
and indeed the function is not considered to be projection-like. However, if one makes the `{a : Set} ->` explicit in the type of member then it is considered so:
```
checkDecl: checking projection-likeness of [RollbackWindow.C]
RollbackWindow.C is abstract or not a function, thus, not considered for projection-likeness
checkDecl: checking projection-likeness of [RollbackWindow.member]
Checking for projection likeness 
member  :  {a : Set} → C a → a → a
  candidates: [(RollbackWindow.C, 1)]
member  :  {a : Set} → C a → a → a
  is projection like in argument 1 for type C
```

### PR DESCRIPTION
**Pull request by UlfNorell**
fixes #7530

### CODE REVIEW

**Review by andreasabel — APPROVED** [review]

**Comment by andreasabel** [comment]
This change apparently triggers unsolved constraints in the `cubical` library: https://github.com/agda/agda/actions/runs/11425039173/job/31786434036?pr=7581#step:8:799
```
Cubical/Categories/Limits/RightKan.agda:109.13-203.41: error: [UnsolvedMetaVariables]
Unsolved metas at the following locations:
  ...
```

**Review by andreasabel — CHANGES_REQUESTED** [review]
Need some investigation why this triggers unsolved constraints.
Possibly a PR to `cubical` is needed before merging this.

**Comment by UlfNorell** [comment]
I believe that what happens is that when more things are projection-like, lossy unification does less work, since a bunch of implicits are now erased and not available for comparison. Regardless, the changes required to cubical are fairly minor.

I'll wait for the cubical PR to be merged then bump the commit here.

**Review by andreasabel — APPROVED** [review]

**Pull request merged**
